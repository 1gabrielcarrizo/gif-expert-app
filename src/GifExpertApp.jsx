

const GifExpertApp = () => {
    return (
        <>
        {/* titulo */}
        
            <h1>GifExpertApp</h1>
            {/* input */}

            {/* listado de gif */}
                {/* gif item */}
        </>
    )
}

export default GifExpertApp