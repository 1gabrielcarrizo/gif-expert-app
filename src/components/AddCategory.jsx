import { useState } from "react"


const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState("")

    const onInputChange = ({target}) => {
        console.log(target.value)
        setInputValue(target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        if(inputValue.trim().length <= 1) return; // no escribe en la lista si es <=1
        setCategories(categories => [...categories, inputValue]) // nos comunicamos con el useState del padre
        setInputValue("") // limpiamos el input
    }

  return (
    <form onSubmit={handleSubmit}>
        <input
        type="text" 
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
        />
    </form>
  )
}

export default AddCategory