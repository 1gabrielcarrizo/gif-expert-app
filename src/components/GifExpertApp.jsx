import { useState } from "react"
import {AddCategory, GifGrid} from "."



const GifExpertApp = () => {

    const [categories, setCategories] = useState([])

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return; // si la categoria esta en la lista, no la agrega de nuevo
        setCategories([newCategory, ...categories]) // caso contrario la agrega con spread operator
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            {/* el componente de abajo es un formulario con un input */}
            <AddCategory
                onNewCategory={onAddCategory} // pasamos la funcion como props
            />

            {/* se muestra el titulo con gifs por cada categoria ingresada*/}
            {
                categories.map((category) => ( // se recorre el array con el "map"
                    <GifGrid
                        key={category}
                        category={category}
                    />
                ))
            }
        </>
    )
}

export default GifExpertApp