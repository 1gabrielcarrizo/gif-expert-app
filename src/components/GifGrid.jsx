import { useEffect, useState } from "react";

import { getGifs } from "../helpers/getGifs";
import GifItem from "./GifItem";

const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]) // para mostrar las imagenes creamos un useState local

    const getImages = async () => { // como no se puede usar el "async" en el useEffect, se crea esta funcion
        const newImages = await getGifs(category);
        setImages(newImages);
    }

    useEffect(() => {
        getImages(); // se llama la funcion asincrona adentro del useEffect
    }, [])


    return (
        <>
            <h3>{ category }</h3>
            <div className="card-grid">
                {
                    images.map(( image ) => (
                        <GifItem
                        key={image.id}
                        {...image} // todas las propiedades que vienen en el image las esparcimos
                        />
                    ))
                }
            </div>
        </>
    )
}

export default GifGrid