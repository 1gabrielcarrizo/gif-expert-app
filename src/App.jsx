import React from 'react'
import { GifExpertApp } from './components'

const App = () => {
  return (
    <GifExpertApp/>
  )
}

export default App