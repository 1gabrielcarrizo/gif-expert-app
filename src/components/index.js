// este es un archivo de barril

import AddCategory from "./AddCategory";
import GifGrid from "./GifGrid";
import GifItem from "./GifItem";
import GifExpertApp from "./GifExpertApp";

export {AddCategory, GifGrid, GifItem, GifExpertApp}