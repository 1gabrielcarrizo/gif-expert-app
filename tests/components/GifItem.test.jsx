import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Pruebas en el componente <GifItem/>', () => {

    // 1 - inicializacion de variables globales
    const title = "Yuri Katsuki"
    const url = "https://yuri-on-ice.com/yuri_katsuki.png"

    test('debe de hacer match con el snapshot', () => {
        // 2 - para hacer la evaluacion del snapshot se debe usar el container
        const { container } = render(<GifItem title={title} url={url} />)
        // 3 - observacion
        expect(container).toMatchSnapshot()
    })

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {

        // 2 - renderizamos nuestro componente
        render(<GifItem title={title} url={url} />)
        // screen.debug()
        // 3 - observacion
        // expect(screen.getByRole('img').src).toBe( url )
        // expect(screen.getByRole('img').alt).toBe( title )
        // podemos simplificar el codigo de la siguiente manera
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('debe de mostrar el titulo en el componente', () => {
        // 2 - renderizamos nuestro componente
        render(<GifItem title={title} url={url} />)
        // 3 - observacion
        expect(screen.getByText(title)).toBeTruthy();
    })
})