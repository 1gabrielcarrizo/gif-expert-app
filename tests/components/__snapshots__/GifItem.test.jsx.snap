// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Yuri Katsuki"
      src="https://yuri-on-ice.com/yuri_katsuki.png"
    />
    <p>
      Yuri Katsuki
    </p>
  </div>
</div>
`;
